language/tools/move-package/src/compilation/compiled_package.rs

```diff
@@ -652,7 +652,65 @@ impl CompiledPackage {
         Ok((compiled_package, CompilationCachingStatus::Recompiled))
     }
 
+    // We take the (restrictive) view that all filesystems are case insensitive to maximize
+    // portability of packages.
+    fn check_filepaths_ok(&self) -> Result<()> {
+        // A mapping of (lowercase_name => (real_name, [info_for_each_occurence])
+        let mut insensitive_mapping = BTreeMap::new();
```

Comment:
Thoughts about a slightly different mapping here? Might make this a bit easier to follow, but I could be wrong.
Just doing `lowercase_file_name => [info_for_each_occurrence]` I think it would make the logic in `errs` seem a bit less wonky for me

Reply:
Totally right. I got it into my head that I needed to group by case-sensitive name, but in reality I don't as long as `[info_for_each_occurence]` is allowed to not be a set. Will update :)